### DISCUSSION TITLE
How to get access to this library

### DISCUSSION THREAD

**Discussion by carlos-prezzee** [DataDog/dd-sdk-kotlin-multiplatform#107] [Q&A]
Hello Datadog team,

Great to see this SDK available! 🎉  

I am wondering how do I get access to the SDK access? 
When creating a new RUM application I don't see a Kotlin Multiplatform option yet.

Also, I have an existing Android & iOS RUM apps. Would the this SDK still connect to those applications? I would I need to create a separate one.

In my current setup I am sharing code using KMP but the UI still native on both. I got a workaround to send logs from KMP to Android but if I can use this SDK directly, it will simplify the integration

Thanks

**Comment by 0xnm** [comment]
Hello @carlos-prezzee! To get an access to the Kotlin Multiplatform SDK you need to open a ticket with [our support](https://www.datadoghq.com/support/), so that we can activate it for your org.

Speaking of reporting, it still depends on your use-case: if you want to completely migrate to KMP SDK it is better to create a dedicated RUM application, but nothing prevents to use this SDK as an addition to the Android / iOS SDKs you already have (the only concern may be only the versioning for the native SDKs being used).

**Reply by carlos-prezzee** [reply]
Thanks @0xnm for the reply. I will send a ticket for it thanks.

When you said the concern is about the native SDKs version it is because it needs to match the expected one from the KMP one right? basically using the native sdk and the kmp will have to match in terms of versions supported?

**Reply by 0xnm** [reply]
Preferably it should match, yes. But if there is a mismatch, it will still work, the main concern is the compilation on iOS in that case.

**Reply by carlos-prezzee** [reply]
Thanks for the advice - will try it out

**Reply by carlos-prezzee** [reply]
@0xnm on the setup when you have already native SDK being initialised in the native parts.
Do you still have to init the KMP SDK as well?

```
// in common source set
 fun initializeDatadog(context: Any? = null) {
     // context should be application context on Android and can be null on iOS
     val appClientToken = <CLIENT_TOKEN>
     val appEnvironment = <ENV_NAME>
     val appVariantName = <APP_VARIANT_NAME>

     val configuration = Configuration.Builder(
             clientToken = appClientToken,
             env = appEnvironment,
             variant = appVariantName
     )
         .build()

     Datadog.initialize(context, configuration, trackingConsent)
 }
 ```

**Reply by 0xnm** [reply]
No, in this case you don't need to. KMP SDK is using native SDKs under the hood.

**Reply by carlos-prezzee** [reply]
@0xnm 
I was trying to test the integration in my existing project but it didn't work. From the documentation it seems that the only possible ways to integrate are either with Cocoapods or embeded frameworks right?

I currently create an XCFramework from my KMP module and ingrate it via SPM to iOS. Is this mode supported?

> output:
ld: warning: ignoring duplicate libraries: '-ldl'
ld: framework 'DatadogObjc' not found
> Task :linkReleaseFrameworkIosArm64 FAILED

Update:

Seems like I am running into this same issue https://github.com/DataDog/dd-sdk-kotlin-multiplatform/issues/94

Following up the conversation there

Cheers
